Tests/SwiftSyntaxTest/SourceLocationConverterTests.swift

```diff
@@ -43,4 +63,87 @@ final class SourceLocationConverterTests: XCTestCase {
     // ```
     _ = SourceLocationConverter(file: "", tree: tree)
   }
+
+  func testSingleSourceLocationDirective() {
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 1)
+      let a = 2
+      """,
+      presumedFile: "other.swift",
+      presumedLine: 1
+    )
+
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 3)
+      let a = 2
+      """,
+      presumedFile: "other.swift",
+      presumedLine: 3
+    )
+
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 4)
+      func foo() {
+      }
+      let a = 2
+      """,
+      presumedFile: "other.swift",
+      presumedLine: 6
+    )
+
+    assertLogicalSourceLocation(
+      """
+      func foo() {
+        print(1)
+      }
+      #sourceLocation(file: "other.swift", line: 1)
+      let a = 2
+      """,
+      presumedFile: "other.swift",
+      presumedLine: 1
+    )
+  }
+
+  func testMultipleSourceLocationDirectives() {
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 10)
+
+      let a = 2
+
+      #sourceLocation(file: "andAnother.swift", line: 20)
+      """,
+      presumedFile: "other.swift",
+      presumedLine: 11
+    )
+
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 10)
+
+      #sourceLocation(file: "andAnother.swift", line: 20)
+
+      let a = 2
+      """,
+      presumedFile: "andAnother.swift",
+      presumedLine: 21
+    )
+  }
+
+  func testResetSourceLocationDirective() {
+    assertLogicalSourceLocation(
+      """
+      #sourceLocation(file: "other.swift", line: 10)
+
+      #sourceLocation()
+
+      let a = 2
+      """,
+      presumedFile: "input.swift",
+      presumedLine: 5
+    )
+  }
 }
```

Comment:
Could you add some tests for negative + hex + other failure cases (eg. string)?

Reply:
I added test case for hex, identifier and string interpolation inside the file name. Everything else should be disallowed by the syntax tree structure.

Reply:
Thanks!